src/node/signer.ts

```diff
@@ -50,34 +55,71 @@ export class TurboNodeSigner extends TurboDataItemAbstractSigner {
     fileStreamFactory,
     fileSizeFactory,
     dataItemOpts,
-  }: TurboFileFactory<NodeFileStreamFactory>): Promise<{
+    events = {
+      onSigningProgress: (event) => {
+        this.logger.debug('Signing progress...', {
+          totalBytes: event.totalBytes,
+          processedBytes: event.processedBytes,
+        });
+      },
+      onSigningError: (error) => {
+        this.logger.error('Signing error...', { error });
+      },
+      onSigningSuccess: () => {
+        this.logger.debug('Signing successful...');
+      },
+    },
+  }: TurboFileFactory<NodeFileStreamFactory> &
```

Comment:
stepping away from the pattern for providing either an emitter or emitter parameters?